hooks/RoboFile.php

```diff
@@ -190,12 +190,30 @@ public function updateDependencies()
      */
     protected function getPatches($module)
     {
-        $path = 'modules/' . $module . '/patches.json';
-        if (file_exists($path)) {
-            return json_decode(file_get_contents($path));
-        } else {
-            return new stdClass();
-        }
+      // Load composer configuration for the given module.
+      $config = json_decode(file_get_contents('modules/' . $module . '/composer.json'));
+
+      // If the module defines patches in composer.json, simply use them.
+      if (!empty($config->extra->{"patches"})) {
+        return $config->extra->{"patches"};
+      }
+
+      // Use the old patch file standard in drupal_tests, which was independent
+      // from any module composer.json patch specifications.
+      $patches_file = 'patches.json';
```

Comment:
👍 for this, we can remove it when Drupal 8.7 or 8.8 is out.

Reply:
Please take care. It follows only the default file name convention, not the old format. If the old/wrong composer patches format should be supported, we would need a different processing here.